manifests/server/config.pp

```diff
@@ -0,0 +1,61 @@
+# This type creates custom configuration files for clickhouse-server.
+#
+# @summary generates xml config from hash via ruby xml-simple
+#
+# @param data
+#   This hash will be converted into xml config placed in `$clickhouse::server::conf_d_dir`.
+#
+#   Root will be `<yandex>` by default.
+# @param section
+#   Should the file be created in configs or users include directory.
+# @param ensure
+#   Subset of attribute `ensure` for `file` type.
+# @param mode
+#   Desired permissions mode for the config file, see `mode` attribute for `file` resource.
+#
+# @example Usage
+#   clickhouse::server::config { 'macros':
+#     data    => {'macros' => [
+#         {'hostname' => ['host.domain.TLD'],}
+#     ]},
+#     section => 'config',
+#   }
+#   #
+#   # Will create file `/etc/clickhouse-server/conf.d/macros.xml`:
+#   # <yandex>
+#   #   <macros>
+#   #     <hostname>host.domain.TLD</hostname>
+#   #   </macros>
+#   # </yandex>
+#
+# @author InnoGames GmbH
+#
+define clickhouse::server::config (
+    Hash                    $data,
+    Enum['config', 'users'] $section,
+    Enum[
+        'present',
+        'file',
```

Comment:
why `file`?

Reply:
because it's valid and desired filetype, simple file https://puppet.com/docs/puppet/5.3/types/file.html#file-attribute-ensure